src/object.d

```diff
@@ -3323,12 +3323,6 @@ unittest
         return _dup!(const(T), T)(a);
 }
 
-/// ditto
-@property T[] dup(T:void)(scope const(T)[] a) @trusted
-{
-    import core.internal.traits : Unconst;
-    return _dup!(const(T), Unconst!T)(a);
```

Comment:
This PR does not attempt to fix dup(), it just simplifies the template bloat. I recommend making a bugzilla issue out of the @safe issue.